repose-aggregator/src/docs/asciidoc/filters/valkyrie-authorization.adoc

```diff
@@ -2,51 +2,224 @@
 
 include::../_includes/in-progress.adoc[]
 
+The Valkyrie Authorization filter performs three distinct functions:
+
+* **Per-Device Authorization**:
+Compares the request method with the user's device permissions to determine whether or not the requested action is allowed.
+* **Response Collection Resource Culling**:
+Values from the origin service's response will be removed based on the user's device permissions.
+Consequently, the user may only view items they have permission to view.
++
+[NOTE]
+====
+Currently, only JSON bodies are supported.
+If you would like to request XML support, please <<../contact-us.adoc#,contact us>>!
+====
+* **Account Permission to Role Translation**:
+Takes the user's account level permissions and translates their names into the standard `X-Roles` header allowing any of the standard means of RBAC to be used on the request.
+This includes translating the user's device permission for the device specified in the `X-Device-Id` header (when the request contains one).
+
+While the authorization function is always performed, the other functions may be enabled or disabled as desired.
```

Comment:
I swear I didn't see a short circuit when I was looking at the code... but I also find this code to be some of the more difficult-to-follow code in our codebase. The `ValkyrieResult` thing seems to be like pretending to have dynamic typing, and it takes me longer to figure out what data is going where. You'd think I would remember the code better by this point..

So if the feature is toggle-able, what is the configuration change to turn it on/off?